lib/matplotlib/pyplot.py

```diff
@@ -200,6 +200,126 @@ def uninstall_repl_displayhook():
 
 draw_all = _pylab_helpers.Gcf.draw_all
 
+_ENSURE_AX_DOC = """
+This function has been decorated by pyplot to have
+an implicit reference to the `plt.gca()` passed as the first argument.
+
+The wrapped function can be called as any of ::
+
+   func(*args, **kwargs)
+   func(ax, *args, **kwargs)
+   func(.., ax=ax)
+
+"""
+
+
+def ensure_ax(func):
+    """Decorator to ensure that the function gets an `Axes` object.
```

Comment:
As I read this line, having not read the source yet, I was expecting this decorator to create a figure/axes pair (`fig, ax = plt.subplots()`) and then pass the axes to the wrapped function.  After reading the source I was initially surprised that this function does not create an axes and it seems like this could result in unexpected behaviour of this decorator.  Is there any utility in allowing this decorator to create an axes via a kwarg to the decorator if one is not passed in?  Though as I type that out it seems like, if this is a reasonable idea, to just have a different decorator that does that?